api-flask/app/database/database.py

```diff
@@ -34,8 +40,15 @@ class AlertsDataBase:
     def __init__(self):
         """Alerts Database initializer."""
         _eng = create_engine(DB_URI)
-        self.session: Session = sessionmaker(_eng)()
-        logger.info("DB connection is initialized.")
+        try:
```

Comment:
@laurentS I can't find a good way to test the db connection. When I do this and the database is down, Python simply hangs and the app doesn't start. Any ideas?

Reply:
I can think of a couple of ways:
- if you're running everything in docker compose, I think you can use the [`healthcheck` ](https://docs.docker.com/compose/compose-file/#healthcheck) option for the `postgres` container, and then `depends_on` with `service_healthy` for your python container. That way, you're sure that postgres is ready to accept connections when python is starting (but it doesn't mean they can talk to each other)
- otherwise, I think I would check access to the port for pg before creating the session. Something like this [SO answer](https://stackoverflow.com/a/19196218/1138710) would allow a quick check at python startup time.
- or maybe the SQLAlchemy [timeout](https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine.params.pool_timeout) option: `create_engine(..., connect_args={"timeout": 30})` before calling `sessionmaker`?

I guess ideally you'd combine all of them so that python code only starts when pg is ready, then at least have a reasonable timeout, and ideally poke the port first to check that connectivity is available?

Reply:
I don't want this to be blocking the app, so the timeout option is actually the only one we need. Thanks for the tips!